src/views/bdsp/encounter-list-view.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "../../components/button.hpp"
+#include "../../utils/general.hpp"
+#include "../pokemon-view.hpp"
+#include "../party-list-view.hpp"
+#include <csight-core.h>
+#include <dmntcht.h>
+#include <functional>
+#include <memory>
+#include <switch.h>
+#include <tesla.hpp>
+
+typedef std::function<std::shared_ptr<csight::Pkx>(size_t slot)> ReadPartyPokemonFn;
+
+class EncounterListView : public tsl::Gui {
+ public:
+  EncounterListView(ReadPartyPokemonFn read_party_slot) : m_read_party_slot(read_party_slot) { }
+
+  virtual tsl::elm::Element *createUI() override {
+    auto frame = new tsl::elm::OverlayFrame("Enconter", " ");
+    auto list = new tsl::elm::List();
+
+    for (int wild_pokemon = 0; wild_pokemon < 6;wild_pokemon++){ // tried to implement gathering of wild party count through offsets but got tired
+      auto current_pokemon = this->m_read_party_slot(wild_pokemon);
+      if (current_pokemon->SpeciesString() != "None") {
+        list->addItem(new PokemonViewButton(current_pokemon->SpeciesString()+ (current_pokemon->IsShiny() ? " ★" : " "), current_pokemon));
+      }
+    }
+
+    frame->setContent(list);
+
+    return frame;
+  }
+
+ private:
+  ReadPartyPokemonFn m_read_party_slot;
+};
+
+class EncounterListViewButton : public Button {
+ public:
+  EncounterListViewButton(ReadPartyPokemonFn read_party_slot) : Button("Encounter") {
+    this->onClick([read_party_slot]() { tsl::changeTo<EncounterListView>(read_party_slot); });
+  }
+};
```

Comment:
CaptureSight needs a generic Pokemon list view, and this is the perfect chance to add it.  

A few small tweaks can be made to make `EncounterListViewButton` generic:
- Rename `ReadPartyPokemonFn` to `ReadPokemonByIndex`
- Rename `EncounterListView` to `PokemonListView`
- Rename `EncounterListViewButton` to `PokemonListViewButton`
- Add a `std::string m_label` class member and use it in `OverlayFrame` so the name of the view can be customized (e.g. "Party" and "Encounters")
- Add a `u32 m_count` class member and use it in the loop so the number of Pokemon can be customized

Putting that together should look like this:
```cpp
typedef std::function<std::shared_ptr<csight::Pkx>(size_t index)> ReadPokemonByIndex;

class PokemonListView : public tsl::Gui {
 public:
  PokemonListView(std::string label, ReadPokemonByIndex read_pokemon_by_index, u32 count) :
      m_label(label), m_count(count), m_read_pokemon_by_index(read_pokemon_by_index) { }

  virtual tsl::elm::Element *createUI() override {
    auto frame = new tsl::elm::OverlayFrame(m_label, " ");
    auto list = new tsl::elm::List();

    for (u32 index = 0; index < m_count; index++) {
      auto current_pokemon = this->m_read_pokemon_by_index(index);
      if (current_pokemon->SpeciesString() != "None") {
        list->addItem(
            new PokemonViewButton(current_pokemon->SpeciesString() + (current_pokemon->IsShiny() ? " ★" : " "), current_pokemon));
      }
    }

    frame->setContent(list);

    return frame;
  }

 private:
  std::string m_label;
  u32 m_count;
  ReadPartyPokemonFn m_read_pokemon_by_index;
};

class PokemonListViewButton : public Button {
 public:
  PokemonListViewButton(std::string label, ReadPokemonByIndex read_pokemon_by_index, u32 count) : Button(label) {
    this->onClick(
        [label, read_pokemon_by_index, count]() { tsl::changeTo<PokemonListView>(label, read_pokemon_by_index, count); });
  }
};
```

Once this PR is merged, I'll reuse this view for the other games as well.